episkos_cli/src/cli.rs

```
@@ -19,12 +19,39 @@ pub enum Commands {
     Create {
         /// Create the file with given data without the interactive mode
         #[arg(short, long)]
-        data: Option<String>,
+        non_interactive: Option<bool>,
+        /// Id of the project
+        #[arg(short, long)]
+        id: Option<String>,
+        /// Directory of the project
+        #[arg(short, long)]
+        directory: Option<Utf8PathBuf>,
+        /// Title of the project
+        #[arg(short, long)]
+        title: Option<String>,
```

I think that you can just put String here, and when title is not given, clap automatically tells the user to use it properly?

---

The default mode (without any flags) is the interactive mode. There is no need to force the user to pass the title as argument with the "title" flag. He will be asked.

---

Oh okay, so basically if no title given => interactive mode?!
Just gotta think about how to handle the user passing other flags, but forgetting the title! If not in a ci environment it could be entered interactively, however that might leed to problems when in a ci env.

Maybe add like a "fail fast" flag for that case? So that the Programm just returns 1 (quits with error) if anything important is missing?

---

In case the `non-interactive` flag is set it will be checked if all necessary data is given. When there is data missing, it will exit with a failure code.

---

Ahh, perfect!